chaincode/contracts/AssetManager.sol

```diff
@@ -0,0 +1,511 @@
+        return _allowances[owner][spender];
+    }
+
+    /** 
+    * @dev `approve` approves an amount of native tokens that the operator is allowed to send on behalf of the `msg.sender` to the `spender`. 
+    * The amount needs to be less than the `globalUserAuthLimit` and the `owner` and `spender` cannot be the zero address.
+    * @param spender The approved recipient of the native tokens.
+    * @param amount The amount of native tokens approved.
+    * @return true if the amount is approved
+    */
+    function approve(address spender, uint256 amount) public whenNotPaused returns (bool) {
+        address owner = msg.sender;
+        require(owner != address(0), "AssetManager: approve from the zero address");
+        require(spender != address(0), "AssetManager: approve to the zero address");
+        require(amount <= globalUserAuthLimit, "AssetManager: approve greater than global limit");
+
+        _allowances[owner][spender] = amount;
+        emit Approval(owner, spender, amount);
+        return true;
+    }
+
+    /** 
+    * @dev `send` is called by an `operator` to send native tokens on behalf of the user. 
+    * It checks the approved amount for the from and to combination and decreases the approved amount by the amount sent.
+    * @param amount The amount of native tokens to send.
+    * @param from The account on whose behalf the operator is sending the native tokens from
+    * @param to The recipient of the native tokens.
+    */
+    function send(uint256 amount, address from, address to) public onlyOperators whenNotPaused() {
+        uint256 balance =  userBalances[from];
+        uint256 currentAllowance = allowance(from, to);
+        if (amount == 0) {
+            revert SendFailed(
+                from,
+                to,
+                amount,
+                balance,
+                currentAllowance,
+                "Send amount cannot equal 0"
+            );
+        }
+        // check from balance
+        if (amount > balance) {
+            revert SendFailed(
+                from,
+                to,
+                amount,
+                balance,
+                currentAllowance,
+                "Insufficient Locked Funds to Send "
+            );
+        }
+        // check from balance
+        if (amount > currentAllowance) {
+            revert SendFailed(
+                from,
+                to,
+                amount,
+                balance,
+                currentAllowance,
+                "Insufficient approved funds to send "
+            );
+        }
+        // withdraw funds from the contract (update userBalance before transfer to protect from reentrancy attack)
+        uint256 newBalance = balance - amount;
+        userBalances[from] = newBalance;
+
+        // update the approved amount.
+        uint256 newLimit = currentAllowance - amount;
+        approve(to,newLimit);
```

Comment:
This would have approved the operator's allowance (as msg.sender), not the user's

Reply:
Great catch 🙏 
Have also updated the tests now which were incorrect, which were mentioned [here](https://github.com/polymorpher/sms-wallet/pull/1#issuecomment-1207398441)